### PR TITLE
Fix Mana Rewards Decay Epoch

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
Companion to iotaledger/iota.go#634.

### CODE REVIEW

File: pkg/protocol/engine/accounts/mana/manager.go [hunk 1]
```diff
@@ -79,24 +79,24 @@ func (m *Manager) GetManaOnAccount(accountID iotago.AccountID, slot iotago.SlotI
 
 	// Apply decay to stored, potential and BIC mana that was calculated when adding the entry to cache
 	// so that it's correct for the requested slot.
-	manaWithDecay, err := manaDecayProvider.ManaWithDecay(mana.Value(), mana.UpdateTime(), slot)
+	DecayManaBySlots, err := manaDecayProvider.DecayManaBySlots(mana.Value(), mana.UpdateTime(), slot)
```
Comment (muXxer) [inline pkg/protocol/engine/accounts/mana/manager.go]: that rename was not intended I guess.

File: pkg/protocol/engine/accounts/mana/manager.go [hunk 2]
```diff
@@ -79,24 +79,24 @@ func (m *Manager) GetManaOnAccount(accountID iotago.AccountID, slot iotago.SlotI
...
 	// Calculate newly generated potential mana since last update time.
-	manaPotential, err := manaDecayProvider.ManaGenerationWithDecay(mana.ExcessBaseTokens(), mana.UpdateTime(), slot)
+	manaPotential, err := manaDecayProvider.GenerateManaAndDecayBySlots(mana.ExcessBaseTokens(), mana.UpdateTime(), slot)
 	if err != nil {
 		return 0, err
 	}
 
 	// Add potential mana to the decayed mana.
-	manaWithDecayAndGeneration, err := safemath.SafeAdd(manaWithDecay, manaPotential)
+	DecayManaBySlotsAndGeneration, err := safemath.SafeAdd(DecayManaBySlots, manaPotential)
```
Comment (muXxer) [inline pkg/protocol/engine/accounts/mana/manager.go]: That rename was not intended I guess.

File: pkg/testsuite/mock/wallet_transactions.go [hunk 3]
```diff
@@ -549,20 +549,12 @@ func (w *Wallet) ClaimDelegatorRewards(transactionName string, inputName string)
...
-	minDeposit := lo.PanicOnErr(api.StorageScoreStructure().MinDeposit(input.Output()))
-
-	if minDeposit > input.BaseTokenAmount() {
-		return 0
-	}
-
-	excessBaseTokens := input.BaseTokenAmount() - minDeposit
-
-	return lo.PanicOnErr(api.ManaDecayProvider().ManaGenerationWithDecay(excessBaseTokens, input.SlotCreated(), w.currentSlot))
+	return lo.PanicOnErr(iotago.PotentialMana(api.ManaDecayProvider(), api.StorageScoreStructure(), input.Output(), input.SlotCreated(), w.currentSlot))
```
Comment (muXxer) [inline pkg/testsuite/mock/wallet_transactions.go:552]: I think the whole wallet is moved to iota.go in this PR: https://github.com/iotaledger/iota-core/pull/585
Maybe you need to fix it there?

**Review by muXxer — CHANGES_REQUESTED** [review]

File: pkg/testsuite/mock/wallet_transactions.go (hunk 3 above)
Comment (PhilippGackstatter) [reply pkg/testsuite/mock/wallet_transactions.go:552]: The wallet itself still exists on develop and it doesn't seem to exist in iota.go. I think only the keymanager was moved or something like that?

Comment (alexsporn) [reply pkg/testsuite/mock/wallet_transactions.go:552]: wallet is still in mock, yes

**Review by muXxer — APPROVED** [review]

**Pull request merged**